### PR TITLE
feat: Add convenience method for constructing key to access account's balance for a given denom

### PR DESCRIPTION
**Pull request by mattverse**
Backport : https://github.com/cosmos/cosmos-sdk/pull/12745

### CODE REVIEW

File: x/bank/legacy/v043/store_test.go [hunk 1]
```diff
@@ -90,13 +90,13 @@ func TestBalanceKeysMigration(t *testing.T) {
 	err = v043bank.MigrateStore(ctx, bankKey, encCfg.Marshaler)
 	require.NoError(t, err)
 
-	newKey := append(types.CreateAccountBalancesPrefix(addr), []byte(fooCoin.Denom)...)
+	newKey := types.CreatePrefixedAccountStoreKey(addr, []byte(fooCoin.Denom))
```
Comment (alexanderbez) [inline x/bank/legacy/v043/store_test.go:93]: IMO the caller shouldn't have to cast into a byte slice -- have `CreatePrefixedAccountStoreKey` do it for them :)

**Review by p0mvn — APPROVED** [review]

File: x/bank/legacy/v043/store_test.go (hunk 1 above)
Comment (p0mvn) [reply x/bank/legacy/v043/store_test.go:93]: Agreed, but should we keep this consistent with upstream to ease future backports / rebases?

Comment (mattverse) [reply x/bank/legacy/v043/store_test.go:93]: This is just test shouldn't matter much right?

Comment (alexanderbez) [reply x/bank/legacy/v043/store_test.go:93]: Is this what core SDK does? Sigh...

**Pull request merged**